### PR TITLE
 port GetINCHI function through api.ts

### PR DESCRIPTION
**Pull request by qur786**

### CODE REVIEW

**Comment by maneetgoyal** [comment]
@qur786 Get in touch with @saikat-samanta and explain the PR to him sometime this week.

**Comment by qur786** [comment]
> @qur786 Get in touch with @saikat-samanta and explain the PR to him sometime this week.

@maneetgoyal ok sir

File: src/api.ts [hunk 1]
```diff
@@ -473,4 +493,32 @@ export function GetStructFromStdINCHI(input: string, options?: GetINCHIOptions):
   return output;
 }
 
+export function GetINCHI(input: INCHIInput): GetINCHIOutput {
+  const InchiAtomArrayType = NAPIArrayType(inchi_Atom, 999);
```
Comment (saikat-samanta) [inline src/api.ts:497]: @qur786 , why length is `999`?

File: src/deref.ts [hunk 2]
```diff
@@ -103,3 +108,46 @@ export function generateINCHIInputV3000(input: ReturnType<typeof inchi_Input_V30
...
+  const ATOMELLenArrayType = NAPIArrayType(refNAPI.types.char, ATOM_EL_LEN);
+  const NUMHIsotopesArrayType = NAPIArrayType(refNAPI.types.char, NUM_H_ISOTOPES + 1);
+  const output: ReturnType<typeof inchi_Atom>[] = input.map((element) => {
+    const { x, y, z, neighbor, bondType, bondStereo, elName, numBonds, numIsoH, isotopicMass, radical, charge } =
+      element;
+    const out: ReturnType<typeof inchi_Atom> = {
+      x: x,
+      y: y,
+      z: z,
+      neighbor: MAXVALArrayType([...neighbor]),
```
Comment (saikat-samanta) [inline src/deref.ts]: @qur786 , What do you think, should we use `new` operator to create an instance?

File: src/api.ts (hunk 1 above)
Comment (qur786) [reply src/api.ts:497]: to create a large array to hold values, It is done by maneet sir,I don't remember the logic behind it now

File: src/deref.ts (hunk 2 above)
Comment (qur786) [reply src/deref.ts]: didn't get it, what do you want to ask?

File: src/api.ts (hunk 1 above)
Comment (saikat-samanta) [reply src/api.ts:497]: @qur786, sir already introduce `MaxValuexyx` in some of the places. Try the same approach

File: src/deref.ts (hunk 2 above)
Comment (saikat-samanta) [reply src/deref.ts]: @qur786, I think this will be `new MAXVALArrayType([...neighbor])`.

File: src/deref.ts [hunk 3]
```diff
@@ -103,3 +108,46 @@ export function generateINCHIInputV3000(input: ReturnType<typeof inchi_Input_V30
...
+      bond_type: MAXVALArrayType([...bondType]),
```
Comment (saikat-samanta) [inline src/deref.ts]:
```suggestion
      bondType: MAXVALArrayType([...bondType]),
```
and so on.

File: src/api.ts (hunk 1 above)
Comment (qur786) [reply src/api.ts:497]: no its just for that instance to create a very large array of inchi_Atom

File: src/deref.ts (hunk 2 above)
Comment (qur786) [reply src/deref.ts]: yeah right saikat , good catch 👍

File: src/deref.ts (hunk 3 above)
Comment (qur786) [reply src/deref.ts]: actually , it is `bond_type` @saikat-samanta , if you look at the type of object we are trying to create there

**Pull request closed without merging**